### DISCUSSION TITLE
Unnecessary http redirection after using trailingSlash

### DISCUSSION THREAD

**Discussion by devv-aman** [gatsbyjs/gatsby#35639] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

We have to redirect all non-trailing slash urls to trailing ones, so to do that, we have use `trailingSlash` as introduced in [4.7](https://github.com/gatsbyjs/gatsby/discussions/34205) with nginx redirection configuration.

But after doing so, there is an unncessary http redirection in between.

![CleanShot 2022-05-12 at 11 37 52](https://user-images.githubusercontent.com/18486164/168003235-e42cc5ff-2496-44c0-9de4-526e3976a6a1.png)

### Reproduction Link

https://gatsby-stage.coinswitch.co/blog

### Steps to Reproduce

Please go to the reproduction link

### Expected Result

Unncessary http redirection should not happen

### Actual Result

Described in the description with screenshot

### Environment

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) arm64 Apple M1 Pro
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.8.0 - /var/folders/ml/l0vnb4dd20bd906kcft81klw0000gp/T/yarn--1652338635927-0.9942688936568478/node
    Yarn: 1.22.17 - /var/folders/ml/l0vnb4dd20bd906kcft81klw0000gp/T/yarn--1652338635927-0.9942688936568478/yarn
    npm: 8.5.5 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 101.0.4951.64
    Safari: 15.4
  npmPackages:
    gatsby: ^4.14.0 => 4.14.0
    gatsby-plugin-client-side-redirect: ^1.1.0 => 1.1.0
    gatsby-plugin-gatsby-cloud: ^4.14.0 => 4.14.0
    gatsby-plugin-google-fonts: ^1.0.1 => 1.0.1
    gatsby-plugin-google-tagmanager: ^3.14.0 => 3.14.0
    gatsby-plugin-hotjar: ^1.1.1 => 1.1.1
    gatsby-plugin-image: ^1.14.2 => 1.14.2
    gatsby-plugin-manifest: ^3.14.0 => 3.14.0
    gatsby-plugin-offline: ^5.14.0 => 5.14.0
    gatsby-plugin-react-helmet: ^4.14.0 => 4.14.0
    gatsby-plugin-robots-txt: ^1.7.0 => 1.7.1
    gatsby-plugin-sass: ^4.14.0 => 4.14.0
    gatsby-plugin-sharp: ^4.5.1 => 4.11.1
    gatsby-plugin-sitemap: ^5.14.0 => 5.14.0
    gatsby-remark-social-cards: ^0.4.1 => 0.4.1
    gatsby-source-filesystem: ^4.14.0 => 4.14.0
    gatsby-source-wordpress: ^6.14.0 => 6.14.0
    gatsby-transformer-sharp: ^4.14.0 => 4.14.0
```

**Comment by LekoArts** [comment]
Hi!

This is something you have to fix on your side with your nginx configuration. It's also possible that the plugin `gatsby-plugin-client-side-redirect` you installed is at fault. I'd also remove that plugin.

**Reply by devv-aman** [reply]
@LekoArts `gatsby-plugin-client-side-redirect` is not used anywhere right now and we're using following nginx configuration

```
rewrite ^([^.]*[^/])$ $1/ permanent;
```

Any suggestions?
